components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -1052,9 +1074,96 @@ void ConversationDriver::PerformAssistantGeneration(
   auto data_completed_callback =
       base::BindOnce(&ConversationDriver::OnEngineCompletionComplete,
                      weak_ptr_factory_.GetWeakPtr(), current_navigation_id);
-  engine_->GenerateAssistantResponse(is_video, page_content, chat_history_,
-                                     input, std::move(data_received_callback),
-                                     std::move(data_completed_callback));
+  bool should_refine_page_content =
+      page_content.length() > GetCurrentModel()
+                                  .options->get_leo_model_options()
+                                  ->max_page_content_length &&
+      input != l10n_util::GetStringUTF8(IDS_AI_CHAT_QUESTION_SUMMARIZE_PAGE) &&
+      features::IsPageContentRefineEnabled();
```

Comment:
feature check should go first in the condition to block possible crashes in new logic

Reply:
fixed in e69fe669d82d317d79eec0a42c2efd68c8eefcad